app/src/main/java/com/SecUpwN/AIMSICD/smsdetection/SmsDetector.java

```diff
@@ -122,372 +116,187 @@ public void run() {
-                            setmsg.setCurrent_lac(mAimsicdService.getCellTracker().getMonitorCell().getLAC());
-                            setmsg.setCurrent_cid(mAimsicdService.getCellTracker().getMonitorCell().getCID());
-                            setmsg.setCurrent_nettype(Device.getNetworkTypeName(mAimsicdService.getCell().getNetType()));
-                            int isRoaming = 0;
-
-                            if("true".equals(mAimsicdService.getCellTracker().getDevice().isRoaming())) {
-                                isRoaming = 1;
-                            }
-                            setmsg.setCurrent_roam_status(isRoaming);
-                            // TODO is this the right place to get upto date geo location?
-                            setmsg.setCurrent_gps_lat(mAimsicdService.lastKnownLocation().getLatitudeInDegrees());
-                            setmsg.setCurrent_gps_lon(mAimsicdService.lastKnownLocation().getLongitudeInDegrees());
-
-                            // Only alert if the timestamp is not in the data base
-                            if(!dbacess.isTimeStampInDB(logcat_timestamp)) {
-                                dbacess.storeCapturedSms(setmsg);
-                                /*dbacess.insertEventLog(
-                                        MiscUtils.getCurrentTimeStamp(),
-                                        mAimsicdService.getCellTracker().getMonitorCell().getLAC(),
-                                        mAimsicdService.getCellTracker().getMonitorCell().getCID(),
-                                        mAimsicdService.getCellTracker().getMonitorCell().getPSC(),
-                                        String.valueOf(mAimsicdService.lastKnownLocation().getLatitudeInDegrees()),
-                                        String.valueOf(mAimsicdService.lastKnownLocation().getLatitudeInDegrees()),
-                                        (int)mAimsicdService.getCell().getAccuracy(),
-                                        3,
-                                        "Detected Type-0 SMS"
-                                );*/
-                                dbacess.toEventLog(3, "Detected Type-0 SMS");
-                                MiscUtils.startPopUpInfo(tContext, 6);
-                            } else {
-                                Log.d(TAG, "Detected Sms already logged");
-                            }
+    private int checkForSms(String line){
+        //0 - null 1 = TYPE0, 2 = MWI, 3 = WAPPUSH
+        for(int i = 0; i < LOADED_DETECTION_STRINGS.length; i++) {
+            //looping thru detection strings to see does logcat line match
+            if(line.contains(LOADED_DETECTION_STRINGS[i].split("#")[0])) {
+
+                if(LOADED_DETECTION_STRINGS[i].split("#")[1].equals("TYPE0")) {
+                    Log.i(TAG, "TYPE0 detected");
+                    return TYPE0;
+                }else if(LOADED_DETECTION_STRINGS[i].split("#")[1].equals("MWI")) {
```

Comment:
`LOADED_DETECTION_STRINGS[i].split("#")[1].equals("MWI")`

Consider this:
LOADED_DETECTION_STRINGS[i].split("#")[1] has value null, no assigned memory. This line will throw NullPointerException and crash application.

If you do instead 
`"MWI".equals(LOADED_DETECTION_STRINGS[i].split("#")[1])
and LOADED_DETECTION_STRINGS[i].split("#")[1]` is still null, this won't crash, and will return false.